graphistry/pygraphistry.py

```diff
@@ -266,25 +320,26 @@ def _handle_auth_url(auth_url, sso_timeout, sso_opt_into_type):
         SSO Login logic.
 
         """
-
-        if in_ipython() or in_databricks() or sso_opt_into_type == 'display':  # If run in notebook, just display the HTML
-            # from IPython.core.display import HTML
-            from IPython.display import display, HTML
-            display(HTML(f'<a href="{auth_url}" target="_blank">Login SSO</a>'))
-            print("Please click the above URL to open browser to login")
-            print(f"If you cannot see the URL, please open browser, browse to this URL: {auth_url}")
-            print("Please close browser tab after SSO login to back to notebook")
-            # return HTML(make_iframe(auth_url, 20, extra_html=extra_html, override_html_style=override_html_style))
-        elif sso_opt_into_type == 'browser':
-            print("Please minimize browser after your SSO login and go back to pygraphistry")
-
-            import webbrowser
-            input("Press Enter to open browser ...")
-            # open browser to auth_url
-            webbrowser.open(auth_url)
-        else:
-            print(f"Please open a browser, browse to this URL, and sign in: {auth_url}")
-            print("After, if you get timeout error, run graphistry.sso_get_token() to complete the authentication")
+        if not in_databricks():
+            # elif in_ipython() or sso_opt_into_type == 'display':  # If run in notebook, just display the HTML
+            if in_ipython() or sso_opt_into_type == 'display':  # If run in notebook, just display the HTML
+
+                from IPython.display import display, HTML
+                display(HTML(f'<a href="{auth_url}" target="_blank">Login SSO</a>'))
+                print("Please click the above URL to open browser to login")
+                print(f"If you cannot see the URL, please open browser, browse to this URL: {auth_url}")
+                print("Please close browser tab after SSO login to back to notebook")
```

Comment:
mixing `display()` and `print()` suggests an incomplete case analysis & handling, can we handle more carefully?

Reply:
we can make print() vs display() options, and sniffers pick the right one